File: src/main/java/me/spartacus04/jext/command/CommandsRegistrant.java
package me.spartacus04.jext.command;

import org.bukkit.plugin.java.JavaPlugin;

public class CommandsRegistrant {

  private static CommandsRegistrant instance = null;
  public static CommandsRegistrant getInstance() {
    if (instance == null) {
      instance = new CommandsRegistrant();
    }
    return instance;
  }

  private CommandsRegistrant() {}

  public void registerCommands(JavaPlugin plugin) {
    new ExecutorDisc().registerTo(plugin);
    new ExecutorDiscGive().registerTo(plugin);
    new ExecutorPlayMusic().registerTo(plugin);
    new ExecutorPlayAt().registerTo(plugin);
    new ExecutorStopMusic().registerTo(plugin);
  }

}

File: src/main/java/me/spartacus04/jext/command/CommandsRegistrant.kt
package me.spartacus04.jext.command

import org.bukkit.plugin.java.JavaPlugin


/**
 * The `CommandsRegistrant` class is a utility class used to register commands.
 *
 * @constructor Class is singleton, so the constructor is private.
 */
internal class CommandsRegistrant private constructor() {
    companion object {
        /**
         * The function `registerCommands` registers all commands to the plugin.
         *
         * @param plugin The plugin instance that you want to register the commands to.
         */
        fun registerCommands(plugin: JavaPlugin) {
            ExecutorDisc().registerTo(plugin)
            ExecutorDiscGive().registerTo(plugin)
            ExecutorPlayMusic().registerTo(plugin)
            ExecutorPlayAt().registerTo(plugin)
            ExecutorStopMusic().registerTo(plugin)
            ExecutorReload(plugin).registerTo(plugin)
            ExecutorJukeboxGui().registerTo(plugin)
            ExecutorFragment().registerTo(plugin)
            ExecutorFragmentGive().registerTo(plugin)

            ExecutorLegacyJukeboxGui(plugin).registerTo(plugin)
        }
    }
}